src/rust/engine/fs/src/directory.rs

```diff
@@ -608,6 +608,32 @@ impl DigestTrie {
     symlinks
   }
 
+  pub fn recursive_entry_counts(&self, path_prefix: &PathBuf) -> HashMap<PathBuf, usize> {
```

Comment:
Should probably make this a `PathBuf`, since the consumer will convert it into one anyway, and `join` returns one.